source/slang/emit.cpp

```diff
@@ -3236,6 +3247,57 @@ struct EmitVisitor
                     }
                     break;
 
+                // We will use the `$X` case as a prefix for
+                // special logic needed when cross-compiling ray-tracing
+                // shaders.
+                case 'X':
+                    {
+                        SLANG_RELEASE_ASSERT(*cursor);
+                        switch(*cursor++)
+                        {
+                        case 'P':
+                            {
+                                // The `$XP` case handles looking up
+                                // the assocaited `location` for a variable
+                                // used as the argument ray payload at a
+                                // trace call site.
+
+                                UInt argIndex = 0;
```

Comment:
The argument the other way is that using a named variable makes it clear that the `0` plays the same role in the assert and the next statement, so that if the argument index ever changed you'd need to change both (or in this case, just change the definition).

Reply:
Fair enough - but perhaps it should be const to make it clearer without further scanning, that it's existence isn't about later mutability.